src/core/transpile.jl

```diff
@@ -699,16 +691,14 @@ function simplify_rx_gate(
 end
 
 
-function cast_to_phase_shift_and_half_rotation_x(gate::Universal;atol=1e-6)
+function cast_to_phase_shift_and_half_rotation_x(gate::Universal, target::Int;atol=1e-6)
     params=get_gate_parameters(gate)
    
-    target=get_connected_qubits(gate)[1]
-
     theta   =params["theta"]
     phi     =params["phi"]
     lambda  =params["lambda"]
 
-    gate_array=Vector{AbstractGate}([])
+    gate_array=Vector{AbstractGateSymbol}([])
```

Comment:
`gate_array=Vector{Gate}([]) ` would enphasize that these are placed Gates

Reply:
Updated